packages/app-mobile/android/app/src/main/res/values/strings.xml

```diff
@@ -1,4 +1,5 @@
 <resources>
     <string name="app_name">Joplin</string>
     <string name="default_notification_channel_id">net.cozic.joplin.notification</string>
+    <string name="no_recent_notes_found">No recent notes found</string>
```

Comment:
If possible it would indeed be better to supply this on the JS side. @fstanis, do you think it's possible?

Reply:
I looked into this and unfortunately it's incredibly tricky. The widget lives in a service which is (in many ways) a separate entity from the main app. This introduces a lot of corner cases, such as the widget being added before the main is initialized, at which point it needs *some* fallback here. Another is the user changing the system language while Joplin isn't running, but having the widget added, in which case the widget has no way to "ask" the main app what the new translation should be (Android handles this automatically via its resource mechanism).

That said, I see the issue with having this translatable bit outside the main app logic. I have a few possible ideas here:

1. Remove the "no recent notes" case all together. The user would see an empty list if there aren't any notes to show.
2. Use an emoji or image instead of text, to indicate the list is empty (any suggestions for what to use?).
3. Get it from the main app on a "best effort" basis. This may mean getting an empty string or having it in the wrong language for some periods of time (per the corner cases above). This is also the hardest to implement IMO.
4. Keep it as-is and translate this string separately using Android's resources (in XML files).
5. Since translations are static, we can have some kind of script that generates Android XML files on build based on the main app's translations. I'm not sure how to bake this into the build process in a good way though.

Reply:
Thanks for looking into it, but if it's too complicated let's leave it as it then. We'll see later if there's a way to translate the string.